internal/handlers/pg/pjson/pjson.go

```diff
@@ -229,8 +298,48 @@ func Unmarshal(data []byte) (any, error) {
 	return fromPJSON(res), nil
 }
 
-// Marshal encodes given built-in or types' package value into pjson.
-func Marshal(v any) ([]byte, error) {
+// Marshal encodes the given document and set its schema in the field $s.
+// Use it when you need to encode a document with schema, for example, when you want to store it in a database.
+func Marshal(d *types.Document) ([]byte, error) {
```

Comment:
I don't really like that we have both `Marshal` and `MarshalElem` here, but both are needed.

`Marshal` is used to make a document for insert and update commands.
`MarshalElem` is used to make a value for `WHERE` clause. 

If you have an idea for a better design, please propose it.